src/sentry/search/snuba/backend.py

```diff
@@ -0,0 +1,352 @@
+from __future__ import absolute_import
+
+import six
+
+import math
+import pytz
+from collections import defaultdict
+from datetime import timedelta, datetime
+
+from django.utils import timezone
+
+from sentry.api.paginator import SequencePaginator
+from sentry.event_manager import ALLOWED_FUTURE_DELTA
+from sentry.models import Release, Group, GroupEnvironment, GroupHash
+from sentry.search.django import backend as ds
+from sentry.utils import snuba
+from sentry.utils.dates import to_timestamp
+
+
+# https://github.com/getsentry/sentry/blob/804c85100d0003cfdda91701911f21ed5f66f67c/src/sentry/event_manager.py#L241-L271
+priority_expr = 'toUInt32(log(times_seen) * 600) + toUInt32(last_seen)'
+
+
+datetime_format = '%Y-%m-%dT%H:%M:%S+00:00'
+
+
+# TODO: Would be nice if this was handled in the Snuba abstraction, but that
+# would require knowledge of which fields are datetimes
+def snuba_str_to_datetime(d):
+    if not isinstance(d, datetime):
+        d = datetime.strptime(d, datetime_format)
+
+    if not d.tzinfo:
+        d = d.replace(tzinfo=pytz.utc)
+
+    return d
+
+
+def calculate_priority_cursor(data):
+    times_seen = sum(data['times_seen'])
+    last_seen = max(int(to_timestamp(snuba_str_to_datetime(d)) * 1000) for d in data['last_seen'])
+    return ((math.log(times_seen) * 600) + last_seen)
+
+
+def _datetime_cursor_calculator(field, fn):
+    def calculate(data):
+        datetimes = [snuba_str_to_datetime(d) for d in data[field]]
+        return fn(int(to_timestamp(d) * 1000) for d in datetimes)
```

Comment:
This might be too clever for what you want to do, but I'm pretty sure you can apply `fn` to `datetimes` instead of the generator expression here.

Reply:
Good call, changed it.